pkg/registry/helpers.go

```diff
@@ -108,3 +104,39 @@ func GetRegistryName(imageName string) (string, error) {
 	}
 	return tag.Registry.Name(), nil
 }
+
+// checkForProxyConfigAndUpdateCert checks if user has configured proxy CA cert data using "TANZU_CLI_PROXY_CA_CERT" environment variable
+// if configured, updates cert data in CertOptions
+func checkForProxyConfigAndUpdateCert(registryCertOpts *CertOptions) error {
+	// check if user provided cert configuration for proxy host, if so, use it
+	proxyCACertData := os.Getenv(constants.ProxyCACert)
+
+	// If proxy CA cert data is available, overwrite the registry cert data
+	err := updateCACertData(proxyCACertData, registryCertOpts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateCACertData(caCertData string, registryCertOpts *CertOptions) error {
+	if caCertData != "" {
+		caCertBytes, err := base64.StdEncoding.DecodeString(caCertData)
+		if err != nil {
+			return errors.Wrap(err, "unable to decode the base64-encoded custom registry CA certificate string")
+		}
+		if len(caCertBytes) != 0 {
+			filePath, err := configpaths.GetRegistryCertFile()
+			if err != nil {
+				return err
+			}
+			err = os.WriteFile(filePath, caCertBytes, 0o644)
+			if err != nil {
+				return errors.Wrapf(err, "failed to write the custom image registry CA cert to file '%s'", filePath)
+			}
+			registryCertOpts.CACertPaths = append(registryCertOpts.CACertPaths, filePath)
+		}
+	}
+	return nil
```

Comment:
I think we need to return nil here because the way the code is written it is this function that ignores an empty ca certificate configuration, which is a valid configuration, for example whenever `TANZU_CLI_PROXY_CA_CERT` is not set.

Reply:
Yes, what @marckhouzam said. Thanks